lib/mastodon/migration_helpers.rb

```diff
@@ -802,6 +806,25 @@ def column_for(table, name)
       columns(table).find { |column| column.name == name }
     end
 
+    # Update the configuration of an index by creating a new one and then
+    # removing the old one
+    def update_index(table_name, index_name, columns, **index_options)
+      if index_name_exists?(table_name, "#{index_name}_old") && index_name_exists?(table_name, index_name)
+        remove_index table_name, index_name
+      elsif index_name_exists?(table_name, index_name)
+        rename_index table_name, index_name, "#{index_name}_old"
+      end
+
+      begin
+        add_index table_name, columns, **index_options.merge(name: index_name, algorithm: :concurrently)
+      rescue ActiveRecord::RecordNotUnique
+        remove_index table_name, name: index_name
+        raise CorruptionError.new(index_name)
+      end
+
+      remove_index table_name, name: "#{index_name}_old" if index_name_exists?(table_name, "#{index_name}_old")
```

Comment:
I'd suggest a slight change so that failure in the migration leaves the database in a state consistent with the one it was in before running the migration:
```suggestion
      if index_name_exists?(table_name, "#{index_name}_new") && index_name_exists?(table_name, index_name)
        remove_index table_name, "#{index_name}_new"
      end

      begin
        add_index table_name, columns, **index_options.merge(name: "#{index_name}_new", algorithm: :concurrently)
      rescue ActiveRecord::RecordNotUnique
        remove_index table_name, name: "#{index_name}_new"
        raise CorruptionError.new(index_name)
      end

      remove_index table_name, name: index_name if index_name_exists?(table_name, index_name)
      rename_index table_name, "#{index_name}_new", index_name
```